File: app/src/main/java/ru/melod1n/vk/api/model/VKPhoto.java
package ru.melod1n.vk.api.model;

import org.json.JSONArray;
import org.json.JSONObject;

import java.io.Serializable;
import java.util.ArrayList;

public class VKPhoto extends VKModel implements Serializable {
    private static final long serialVersionUID = 1L;

    private int id;
    private int albumId;
    private int ownerId;
    private String text;
    private int date;
    private ArrayList<VKPhotoSize> sizes;
    private int width;
    private int height;

    public VKPhoto(JSONObject o) {
        setId(o.optInt("id", -1));
        setAlbumId(o.optInt("album_id", -1));
        setOwnerId(o.optInt("owner_id", -1));
        setText(o.optString("text"));
        setDate(o.optInt("date"));

        JSONArray oSizes = o.optJSONArray("sizes");
        if (oSizes != null) {
            ArrayList<VKPhotoSize> sizes = new ArrayList<>();
            for (int i = 0; i < oSizes.length(); i++) {
                sizes.add(new VKPhotoSize(oSizes.optJSONObject(i)));
            }

            setSizes(sizes);
        }

        setWidth(o.optInt("width"));
        setHeight(o.optInt("height"));
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getAlbumId() {
        return albumId;
    }

    public void setAlbumId(int albumId) {
        this.albumId = albumId;
    }

    public int getOwnerId() {
        return ownerId;
    }

    public void setOwnerId(int ownerId) {
        this.ownerId = ownerId;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public int getDate() {
        return date;
    }

    public void setDate(int date) {
        this.date = date;
    }

    public ArrayList<VKPhotoSize> getSizes() {
        return sizes;
    }

    public void setSizes(ArrayList<VKPhotoSize> sizes) {
        this.sizes = sizes;
    }

    public int getWidth() {
        return width;
    }

    public void setWidth(int width) {
        this.width = width;
    }

    public int getHeight() {
        return height;
    }

    public void setHeight(int height) {
        this.height = height;
    }
}

File: app/src/main/java/ru/melod1n/vk/api/model/VKPhoto.kt
package ru.melod1n.vk.api.model

import org.json.JSONObject
import java.io.Serializable
import java.util.*

class VKPhoto(o: JSONObject) : VKModel(), Serializable {
    var id = 0
    var albumId = 0
    var ownerId = 0
    var text: String? = null
    var date = 0
    var sizes: ArrayList<VKPhotoSize>? = null
    var width = 0
    var height = 0

    companion object {
        private const val serialVersionUID = 1L
    }

    init {
        id = o.optInt("id", -1)
        albumId = o.optInt("album_id", -1)
        ownerId = o.optInt("owner_id", -1)
        text = o.optString("text")
        date = o.optInt("date")
        val oSizes = o.optJSONArray("sizes")
        if (oSizes != null) {
            var sizes = ArrayList<VKPhotoSize>()
            for (i in 0 until oSizes.length()) {
                sizes.add(VKPhotoSize(oSizes.optJSONObject(i)))
            }
            sizes = sizes
        }
        width = o.optInt("width")
        height = o.optInt("height")
    }
}